app/controllers/users_controller.rb

```diff
@@ -1792,6 +1789,53 @@ def user_menu_bookmarks
     }
   end
 
+  def user_menu_messages
```

Comment:
I think this endpoint needs the `SiteSetting.enable_person_messages && is_staff` check as well.